client/src/components/Overview/ImageGallery.jsx

```diff
@@ -3,21 +3,38 @@ import ImageGalleryThumbnailColumn from './ImageGalleryThumbnailColumn.jsx';
 import ImageGalleryForwardArrow from './ImageGalleryForwardArrow.jsx';
 import ImageGalleryBackArrow from './ImageGalleryBackArrow.jsx';
 
-var ImageGallery = function({ productStyles }) {
-  let photo = productStyles.results[0].photos[0].url;
-  return (
-    <div className="image-gallery-container" data-testid="image-gallery">
-      <div className="image-gallery">
-        <img className="ig-main-image" alt="Main product gallery image" src={photo} />
-        <ImageGalleryBackArrow />
-        <ImageGalleryForwardArrow />
-        <ImageGalleryThumbnailColumn productStyles={productStyles} />
-        <div className="image-gallery-fullscreen-toggle">
-          <span className="material-icons">fullscreen</span>
+class ImageGallery extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      selectedStyle: 0, // this will later move up to the overview component, but for quick testing, it's here.
+      selectedPhoto: 0
+    };
+    this.thumbnailClicked = this.thumbnailClicked.bind(this);
+  };
+
+  thumbnailClicked(event) {
+    this.setState({ selectedPhoto: event.target.dataset.index});
+  }
+
+  render() {
+    let { selectedStyle, selectedPhoto } = this.state;
+    let { productStyles } = this.props;
+    let photo = productStyles.results[selectedStyle].photos[selectedPhoto].url;
+    return (
+      <div className="image-gallery-container" data-testid="image-gallery">
+        <div className="image-gallery">
+          <img className="ig-main-image" src={photo} />
+          <ImageGalleryBackArrow />
+          <ImageGalleryForwardArrow />
+          <ImageGalleryThumbnailColumn productStyles={productStyles} thumbnailClicked={this.thumbnailClicked} selected={this.state.selectedPhoto}/>
```

Comment:
this is now a nit, but you could refactor the "bind" above into an arrow/anonymous function here:

`thumbnailClicked={(event) => this.thumbnailClicked(event)}`

Reply:
This maintains the `this` of this component and is just a little more concise